File: app/src/main/java/com/sermage/mymoviecollection/adapters/TrailerAdapter.java
package com.sermage.mymoviecollection.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.sermage.mymoviecollection.R;
import com.sermage.mymoviecollection.pojo.Trailers;

import java.util.ArrayList;
import java.util.List;

public class TrailerAdapter extends RecyclerView.Adapter<TrailerAdapter.TrailerHolder> {

    private List<Trailers> trailers;
    private OnTrailerClickListener trailerClickListener;

    public TrailerAdapter() {
        trailers=new ArrayList<>();
    }

    public interface OnTrailerClickListener{
        void onTrailerClick(String url);
    }

    public void setTrailerClickListener(OnTrailerClickListener trailerClickListener) {
        this.trailerClickListener = trailerClickListener;
    }

    @NonNull
    @Override
    public TrailerHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view= LayoutInflater.from(parent.getContext()).inflate(R.layout.trailer_item,parent,false);
        return new TrailerHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull TrailerHolder holder, int position) {
        holder.textViewNameOfTrailer.setText(trailers.get(position).getName());

    }

    @Override
    public int getItemCount() {
        return trailers.size();
    }

    class TrailerHolder extends RecyclerView.ViewHolder{

        TextView textViewNameOfTrailer;
        private static final String YOUTUBE_URL="https://www.youtube.com/watch?v=";

        public TrailerHolder(@NonNull View itemView) {
            super(itemView);
            textViewNameOfTrailer=itemView.findViewById(R.id.textViewNameOfTrailer);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    trailerClickListener.onTrailerClick(YOUTUBE_URL+trailers.get(getAdapterPosition()).getKey());
                }
            });
        }
    }

    public void setTrailers(List<Trailers> trailers) {
        this.trailers = trailers;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/sermage/mymoviecollection/adapters/TrailerAdapter.kt
package com.sermage.mymoviecollection.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.sermage.mymoviecollection.R
import com.sermage.mymoviecollection.pojo.Trailers
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.trailer_item.view.*


class TrailerAdapter : RecyclerView.Adapter<TrailerAdapter.TrailerHolder>() {
    inner class TrailerHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textViewNameOfTrailer = itemView.textViewNameOfTrailer
        val imageViewPlay = itemView.imageViewPlay
    }

    private val YOUTUBE_URL = "https://www.youtube.com/watch?v="
    var onTrailerClickListener: OnTrailerClickListener? = null

    var trailers = listOf<Trailers>()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TrailerHolder {
        val view: View =
            LayoutInflater.from(parent.context).inflate(R.layout.trailer_item, parent, false)
        return TrailerHolder(view)
    }

    override fun onBindViewHolder(holder: TrailerHolder, position: Int) {
        holder.textViewNameOfTrailer.text = trailers[position].name
        val path = YOUTUBE_URL + trailers[position].key
        Picasso.get().load("https://img.youtube.com/vi/${trailers[position].key}/1.jpg")
            .into(holder.imageViewPlay)
        holder.itemView.setOnClickListener {
            onTrailerClickListener?.onTrailerClick(path)
        }
    }


    override fun getItemCount(): Int {
        return trailers.size
    }

    interface OnTrailerClickListener {
        fun onTrailerClick(url: String?)
    }

}

